tests/extensions/test_version.py

```diff
@@ -0,0 +1,210 @@
+"""Tests for pystac.extensions.version."""
+
+import datetime
+import unittest
+
+import pystac
+from pystac.extensions import version
+
+URL_TEMPLATE = 'http://example.com/catalog/%s.json'
+
+
+def MakeItem(year):
+    """Create basic test items that are only slightly different."""
+    asset_id = f'USGS/GAP/CONUS/{year}'
+    start = datetime.datetime(year, 1, 2)
+
+    item = pystac.Item(id=asset_id, geometry=None, bbox=None, datetime=start, properties={})
+    item.set_self_href(URL_TEMPLATE % year)
+
+    item.ext.enable(pystac.Extensions.VERSION)
+
+    return item
+
+
+class VersionItemExtTest(unittest.TestCase):
+    version = '1.2.3'
+
+    def setUp(self):
+        super().setUp()
+        self.item = MakeItem(2011)
+
+        self.item.ext.enable(pystac.Extensions.VERSION)
+
+    def tearDown(self):
+        super().tearDown()
+        self.item.validate()
```

Comment:
No, this is bad.  I was trying to avoid so much duplication in the tests.  Looking at the docs for tearDown, I shouldn't do this.  https://docs.python.org/3/library/unittest.html#unittest.TestCase.tearDown

> Method called immediately after the test method has been called and the result recorded.